src/native/containers/dn-simdhash-specialization.h

```diff
@@ -0,0 +1,524 @@
+			uint32_t value_slot_index = (bucket_index * DN_SIMDHASH_BUCKET_CAPACITY) + new_index;
+			// We found a bucket with space, so claim the first free slot
+			dn_simdhash_bucket_set_count(bucket_address->suffixes, new_index + 1);
+			dn_simdhash_bucket_set_suffix(bucket_address->suffixes, new_index, suffix);
+			bucket_address->keys[new_index] = key;
+			*address_of_value(buffers, value_slot_index) = value;
+			// printf("Inserted [%zd, %zd] in bucket %d at index %d\n", key, value, bucket_index, new_index);
+			// If we cascaded out of our original target bucket, scan through our probe path
+			//  and increase the cascade counters. We have to wait until now to do that, because
+			//  during the process of getting here we may end up finding a duplicate, which would
+			//  leave the cascade counters in a corrupted state
+			adjust_cascaded_counts(buffers, first_bucket_index, bucket_index, 1);
+			return DN_SIMDHASH_INSERT_OK;
+		}
+
+		// The current bucket is full, so try the next bucket.
+	END_SCAN_BUCKETS(first_bucket_index, bucket_index, bucket_address)
+
+	return DN_SIMDHASH_INSERT_NEED_TO_GROW;
+}
+
+static void
+DN_SIMDHASH_REHASH_INTERNAL (DN_SIMDHASH_T_PTR hash, dn_simdhash_buffers_t old_buffers)
+{
+	BEGIN_SCAN_PAIRS(old_buffers, key_address, value_address)
+		uint32_t key_hash = DN_SIMDHASH_KEY_HASHER(DN_SIMDHASH_GET_DATA(hash), *key_address);
+		// This theoretically can't fail, since we just grew the container and we
+		//  wrap around to the beginning when there's a collision in the last bucket.
+		dn_simdhash_insert_result ok = DN_SIMDHASH_TRY_INSERT_INTERNAL(
+			hash, *key_address, key_hash,
+			*value_address,
+			0
+		);
+		// FIXME: Why doesn't assert(ok) work here? Clang says it's unused
+		if (ok != DN_SIMDHASH_INSERT_OK)
+			assert(0);
+	END_SCAN_PAIRS(old_buffers, key_address, value_address)
+}
+
+#if DN_SIMDHASH_HAS_REMOVE_HANDLER
+static void
+DN_SIMDHASH_DESTROY_ALL (DN_SIMDHASH_T_PTR hash)
+{
+	dn_simdhash_buffers_t buffers = hash->buffers;
+	BEGIN_SCAN_PAIRS(buffers, key_address, value_address)
+		DN_SIMDHASH_ON_REMOVE(DN_SIMDHASH_GET_DATA(hash), *key_address, *value_address);
+	END_SCAN_PAIRS(buffers, key_address, value_address)
+}
+#endif
+
+
+// TODO: Store this by-reference instead of inline in the hash?
+dn_simdhash_vtable_t DN_SIMDHASH_T_VTABLE = {
+	DN_SIMDHASH_REHASH_INTERNAL,
+#if DN_SIMDHASH_HAS_REMOVE_HANDLER
+	DN_SIMDHASH_DESTROY_ALL,
+#else
+	NULL,
+#endif
+};
+
+
+#ifndef DN_SIMDHASH_NO_DEFAULT_NEW
+DN_SIMDHASH_T_PTR
+DN_SIMDHASH_NEW (uint32_t capacity, dn_allocator_t *allocator)
+{
+	// If this isn't satisfied, the generic code will allocate incorrectly sized buffers
+	// HACK: Use static_assert because for some reason assert produces unused variable warnings only on CI
+	struct silence_nuisance_msvc_warning { bucket_t a, b; };
+	static_assert(
+		sizeof(struct silence_nuisance_msvc_warning) == (sizeof(bucket_t) * 2),
+		"Inconsistent spacing/sizing for bucket_t"
+	);
```

Comment:
what was the warning.  can you just add `(void)allocator;` in the body?

Reply:
The MSVC warning was something like 'anonymous struct defined in expression' for the static_assert

Reply:
oh. did you try something like `static_assert (sizeof(struct { bucket_t a, b; }) == sizeof(bucket_t) * 2);` ?

Isn't that always true?  if the compiler needs to add padding in order to ensure that arrays of `bucket_t` are properly aligned, that will be included in `sizeof(bucket_t)`, I think.

Reply:
I was originally doing something with a temporary array of 2 elements, but it wouldn't let me assert on that either. Maybe I'll just remove this assertion then since it sounds like it's not doing anything useful.